src/function/table/call/hnsw/query_hnsw_index.cpp

```diff
@@ -78,9 +137,18 @@ static common::offset_t tableFunc(const TableFuncInput& input, TableFuncOutput&
             bindData->boundInput.indexEntry->getConfig().copy());
         index->setUpperEntryPoint(bindData->boundInput.indexEntry->getUpperEntryPoint());
         index->setLowerEntryPoint(bindData->boundInput.indexEntry->getLowerEntryPoint());
+
+        auto& indexColumnType =
```

Comment:
These access pattern is a bit convoluted.

An index entry should know its type. And dimension should be accessed through ArrayType::getDimention(LogicalType)

Reply:
> An index entry should know its type.

Hmm, I'm not sure about this. The dependency between index and the table column should be enough to provide such info.

> And dimension should be accessed through ArrayType::getDimention(LogicalType)

Nice catch. Thanks!